TempMail/utils/Email.swift

```diff
@@ -49,26 +74,69 @@ class Email : ObservableObject{
                     return
                 }
                 guard let data = data else {return}
-                guard let inbox = self.parseInboxData(data: data) else {return}
+                let info = self.parseData(data: data, type: "inbox")
                 //updates instance variable
                 DispatchQueue.main.async {
                     if self.email_addr == email_addr {
-                        self.inbox = inbox
+                        self.inbox = info.inbox
                     }
                 }
             }.resume()
         }
     }
     
-    //parses Inbox JSON data
-    private func parseInboxData(data: Data) -> [InboxModel]? {
-        var response: [InboxModel]?
-        do {
-            response = try JSONDecoder().decode([InboxModel].self, from: data)
-        } catch {
-            print("Fetch failed: \(error.localizedDescription)")
+    //fetches message data
+    private func fetchMessage(id: Int) -> MessageModel? {
+        if msgs![id] != nil {
+            return self.msgs![id]
         }
-        return response
+        guard let url = URL(string: "https://www.1secmail.com/api/v1/?action=readMessage&login=\(self.email_addr)&domain=1secmail.com&id=\(id)") else {fatalError("Invalid URL")}
+        //makes the request to server
+        URLSession.shared.dataTask(with: url) { (data, response, error) in
+            if let error = error {
+                print("Error: \(error)")
+                return
+            }
+            guard let data = data else {return}
+            let info = self.parseData(data: data, type: "msg")
+            //updates instance variable
+            DispatchQueue.main.async {
+                self.msgs![id] = info.msg
+                //download pics
+                let attachments = self.msgs![id]?.attachments
+                
+                if attachments != nil {
+                    for element in attachments! {
+                        if element.filename.range(of: ".png", options: .caseInsensitive) != nil || element.filename.range(of: ".jpg",options: .caseInsensitive) != nil || element.filename.range(of: ".jpeg",options: .caseInsensitive) != nil {
+                            print("in here")
+                            self.downloadAttachment(id: id, attachment: element)
+                        }
+                    }
+                }
+            }
+        }.resume()
+        
+        return self.msgs![id]
```

Comment:
Since dataTask is async, wouldn't it be possible to return self.msgs![id] before its been retrieved? Can we move the return statement to the main thread in the async block or does swift complain about that? If it does complain, they maybe we need to have a similar fetch and get idea like with the inbox.